### PR TITLE
Begin "vehicleSpeed" refactoring. Module switched to "FunctionalPointerSensor" class. #3106 #3107 #3108.

### PR DESCRIPTION
**Pull request by alxrMironov**

### CODE REVIEW

File: firmware/console/status_loop.cpp [hunk 1]
```diff
@@ -689,13 +689,13 @@ void updateTunerStudioState(TunerStudioOutputChannels *tsOutputChannels DECLARE_
...
 #if EFI_VEHICLE_SPEED
-	float vehicleSpeed = getVehicleSpeed();
+	float vehicleSpeed = Sensor::get(SensorType::VehicleSpeed).value_or(0);
 	tsOutputChannels->vehicleSpeedKph = vehicleSpeed;
 	tsOutputChannels->speedToRpmRatio = vehicleSpeed / rpm;
 
 #endif /* EFI_VEHICLE_SPEED */
 #endif /* EFI_PROD_CODE */
-
+
```
Comment (mck1117) [inline firmware/console/status_loop.cpp:698]: stray tab

File: firmware/controllers/algo/launch_control.cpp [hunk 2]
```diff
@@ -55,7 +55,7 @@ bool LaunchControlBase::isInsideSwitchCondition() const {
  * then we have to return true, and trust that we would disable by other condition!
  */ 
 bool LaunchControlBase::isInsideSpeedCondition() const {
-	int speed = getVehicleSpeed(PASS_ENGINE_PARAMETER_SIGNATURE);
+	int speed = static_cast<int>(Sensor::get(SensorType::VehicleSpeed).value_or(0));
```
Comment (mck1117) [inline firmware/controllers/algo/launch_control.cpp:58]: no explicit cast required, perfectly safe to just assign a float to an int

Comment (mck1117) [reply firmware/controllers/algo/launch_control.cpp:58]: on second thought, why is `speed` an int at all? Just make it a float.

File: firmware/init/sensor/init_vehicle_speed_sensor.cpp [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+};
+
+static GetVehicleSpeedWrapper vehicleSpeedWrapper;
+
+static FunctionPointerSensor vehicleSpeedSensor(SensorType::VehicleSpeed,
+[]() {
+	return vehicleSpeedWrapper.getVehicleSpeed();
+});
+
+void initVehicleSpeedSensor(DECLARE_ENGINE_PARAMETER_SIGNATURE) {
```
Comment (mck1117) [inline firmware/init/sensor/init_vehicle_speed_sensor.cpp:22]: I don't see a call to `initVehicleSpeedSensor`, so the sensor never gets registered!

**Comment by mck1117** [comment]
looks like this:
- fully does #3107 
- fully does #3108

File: firmware/controllers/algo/launch_control.cpp (hunk 2 above)
Comment (alxrMironov) [reply firmware/controllers/algo/launch_control.cpp:58]: im will remove cast and make speed type changing with another PR and commit. Ok?

File: firmware/init/sensor/init_vehicle_speed_sensor.cpp (hunk 3 above)
Comment (rusefillc) [reply firmware/init/sensor/init_vehicle_speed_sensor.cpp:22]: @alxrMironov ^^^ this one is probably the most important one to get this PR less red more green?

**Review by mck1117 — APPROVED** [review]

**Pull request merged**